src/lib/containers/FilePreview/FileEntityPreview.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import {
+  EntityBundle,
+  isFileEntity,
+  FileHandleAssociateType,
+  implementsCloudProviderFileHandleInterface,
+} from '../../utils/synapseTypes'
+import FileHandleContentRenderer from './FileHandleContentRenderer'
+import { PreviewRendererType } from './PreviewRendererType'
+import {
+  getOriginalFileHandleRenderer,
+  getPreviewFileHandleRenderer,
+} from './PreviewUtils'
+import { useSynapseContext } from '../../utils/SynapseContext'
+import { Alert } from 'react-bootstrap'
+import { SignInPrompt } from '../ErrorBanner'
+
+export type FileEntityPreviewProps = {
+  /** An entity bundle containing a FileEntity, and its fileHandles */
+  bundle: EntityBundle
+}
+
+/**
+ * Given an entity bundle that contains a FileEntity, render a preview of the file. The rendered preview will be determined based on the
+ * content type and file name of the file handle(s) associated with the FileEntity.
+ * @param props
+ */
+export default function FileEntityPreview(props: FileEntityPreviewProps) {
+  const { bundle } = props
+  const { accessToken } = useSynapseContext()
+  const isLoggedIn = !!accessToken
+
+  if (!isFileEntity(bundle.entity)) {
+    throw new Error(`Entity ${bundle.entity.id!} is not a FileEntity`)
```

Comment:
Let's wrap all Markdown widgets in an error boundary (coming in next PR). There are too many places where things can go wrong in each layer, and it doesn't make sense to duplicate error handling logic in all of those places. Much easier to just throw an Error.

Reply:
I agree